modmesh/pilot/_profiling.py

```diff
@@ -136,4 +149,189 @@ def key_func(d): d["total_time"]
         self._table.showMaximized()
         self._table.show()
 
+
+class ProfileConfigWidget(QWidget):
+    def __init__(self, app, parent=None):
+        super().__init__(parent)
+        self.app = app
+
+    def init_ui(self, options):
+        main_layout = QVBoxLayout(self)
+
+        def on_check_change(check_state, opt, item):
+            if check_state == Qt.Checked:
+                self.app.add_opt(opt, item)
+            else:
+                self.app.rm_opt(opt, item)
+
+        for opt in options:
+            title = opt[0]
+            choices = opt[1]
+
+            vbox = QVBoxLayout()
+            vbox.setStretch(0, 4)
+
+            title_label = QLabel(title)
+            vbox.addWidget(title_label)
+
+            for c in choices:
+                checkbox = QCheckBox(c)
+                checkbox.setTristate(False)
+                cbfunc = functools.partial(on_check_change, opt=title, item=c)
+                checkbox.checkStateChanged.connect(cbfunc)
+                vbox.addWidget(checkbox)
+
+            main_layout.addLayout(vbox)
+
+        cmd_vbox = QVBoxLayout()
+        cmd_vbox.setStretch(0, 4)
+
+        command_label = QLabel("Command")
+        cmd_vbox.addWidget(command_label)
+
+        run_button = QPushButton("run")
+        run_button.clicked.connect(self.app.run_profiling)
+        cmd_vbox.addWidget(run_button)
+
+        main_layout.addLayout(cmd_vbox)
+
+        self.setLayout(main_layout)
+
+
+class RunProfiling(PilotFeature):
+    def __init__(self, *args, **kw):
+        super().__init__(*args, **kw)
+        self._diag = QtWidgets.QFileDialog()
+        self._diag.setFileMode(QtWidgets.QFileDialog.ExistingFile)
+        self._diag.setWindowTitle("Open profiling file")
+        self.module = None
+        self.prof_candidate = {}
+        self.prof_opt = {}
+
+    def populate_menu(self):
+        self._add_menu_item(
+            menu=self._mgr.profilingMenu,
+            text="SimpleArray Arithmetic Profiling",
```

Comment:
The description should be something like "Profiling script".  The tip should be changed similarly.